sklearn/metrics/cluster/tests/test_common.py

```diff
@@ -174,7 +176,7 @@ def generate_formats(y):
             np.array([str(x) + "-a" for x in y.tolist()], dtype=object),
             "array of strs",
         )
-        yield y - 1, "including negative ints"
+        yield [x - 2 if x != 1 else x - 100 for x in y], "including negative ints"
```

Comment:
Yes, it is, if we want the new scoring function to undergo the same common set of tests as all other metrics do.

DBCV intentionally violates (label) format invariance (as would any other density based metric that performs dedicated handling for noise points) for one particular negative value (namely `-1`) because it's used per covention to indicate noise points, which then get treated differently from other labels.

My modification is intended to shift any negative inputs to avoid values of `-1`, which would cause failures. Since DBCV is the first sklearn clustering metric to incorporate the concept of noise, this hasn't been an issue before.

Tbh I anticipated this raising some eyebrows when I first checked it in, but it's the most favorable, pragmatic option for now IMO.
I at least added an explanatory comment:  b92b813d11a7740dc4d55cc6259fe4727a49a109
I can either remove DBCV from common testing entirely or leave it as is.

An overhaul of the common cluster metrics test suite to generally accomodate density based metrics is out of scope for this change IMO. I would happily look into this at a later point in the form of a dedicated issue/PR though. (Especially since we've seen some interest in adding further density based metrics, this could be a valuable change for me to pursue to enable upcoming contributions by other folks)